src/patternfly/layouts/Toolbar/styles.scss

```diff
@@ -26,20 +33,25 @@
 // Section
 .pf-l-toolbar__section {
   flex-basis: 100%;
-
-  // Add margin-top for stacked sections
-  &:not(:first-child) {
-    margin-top: var(--pf-l-toolbar__section--MarginTop);
-  }
 }
 
 // Group
-.pf-l-toolbar__group:not(:last-child) {
-  margin-right: var(--pf-l-toolbar__group--MarginRight);
+.pf-l-toolbar__group {
+  padding: var(--pf-global--spacer--xl);
+
+  // Add margin-top for stacked groups
+  margin-top: var(--pf-l-toolbar__section--MarginTop);
```

Comment:
this applies a `margin-top` to all toolbar groups. You only want there to be vertical space between groups when they wrap and are stacked on top of one another.

Also you're using the variable `--pf-l-toolbar__section--MarginTop` for `.pf-l-toobar__group`. You would want to create `--pf-l-toolbar__group--MarginTop` and use that here instead.